sdk/python/lib/pulumi/provider/experimental/host.py

```diff
@@ -17,13 +17,12 @@
 from typing import Optional
 
 from ...provider import main
-from .metadata import Metadata
 from .provider import ComponentProvider
 
 is_hosting = False
 
 
-def component_provider_host(metadata: Optional[Metadata] = None):
+def component_provider_host(name: str, namespace: Optional[str]):
```

Comment:
I *think* you need to provide a default value so the user doesn't have to explicitly call it with `None`:
```suggestion
def component_provider_host(name: str, namespace: Optional[str] = None):
```